ipinfo/cmd_init.go

```diff
@@ -0,0 +1,216 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/spf13/pflag"
+	"golang.org/x/term"
+)
+
+func printHelpInit() {
+	fmt.Printf(
+		`Usage: %s init [<opts>] [<token>]
+
+Options:
+  --token <tok>, -t <tok>
+    token to login with.
+    (this is potentially unsafe; let the CLI prompt you instead).
+  --no-check
+    disable checking if the token is valid or not.
+    default: false.
+  --help, -h
+    show help.
+`, progBase)
+}
+
+type signupCli struct {
+	SignupURL string `json:"signupURL"`
+}
+
+type tokenCli struct {
+	Token string `json:"token"`
+}
+
+func cmdInit() error {
+	var num int
+	var fTok string
+	var fNoCheck bool
+
+	pflag.StringVarP(&fTok, "token", "t", "", "the token to save.")
+	pflag.BoolVar(&fNoCheck, "no-check", false, "disable checking if token is valid.")
+	pflag.BoolVarP(&fHelp, "help", "h", false, "show help.")
+	pflag.Parse()
+
+	if fHelp {
+		printHelpInit()
+		return nil
+	}
+
+	// get args without subcommand.
+	args := pflag.Args()[1:]
+
+	// only token arg allowed.
+	if len(args) > 1 {
+		return errors.New("invalid arguments")
+	}
+
+	// allow only flag or arg for token but not both.
+	if fTok != "" && len(args) > 0 {
+		return errors.New("ambiguous token input source")
+	}
+
+	// get token, from flag or command line.
+	tok := fTok
+	if len(args) > 0 {
+		tok = args[0]
+	}
+	if tok != "" && !fNoCheck {
```

Comment:
This is not taking into account the situation where:

1. User provided token somehow.
2. Did `--no-check`.

This will cause it to proceed incorrectly into the step phase.